packages/x-components/src/x-modules/facets-next/store/__tests__/getters.spec.ts

```diff
@@ -0,0 +1,194 @@
+    });
+
+    it('returns an object containing all the filters indexed by its facet', () => {
+      const store = createFacetsStore([
+        createNextSimpleFilter('color', 'Red', false),
+        createNextSimpleFilter('color', 'Blue', true),
+        createNextHierarchicalFilter('category', 'Summer', false),
+        createNextHierarchicalFilter('category', 'Shorts', true),
+        createNextNumberRangeFilter('price', { min: 0, max: 25 }, false),
+        createNextNumberRangeFilter('price', { min: 25, max: 50 }, true),
+        createNextEditableNumberRangeFilter('age', { min: null, max: 5 }),
+        createNextEditableNumberRangeFilter('size', { min: null, max: null }),
+        // Raw filters don't belong to a facet so they're not retrieved by this getter
+        createRawFilter('size:xl')
+      ]);
+
+      expect(store.getters.filtersByFacet).toEqual({
+        color: [store.state.filters['color:Red'], store.state.filters['color:Blue']],
+        category: [store.state.filters['category:Summer'], store.state.filters['category:Shorts']],
+        price: [store.state.filters['price:0-25'], store.state.filters['price:25-50']],
+        age: [store.state.filters['age:*-5']],
+        size: [store.state.filters['size:*-*']]
+      });
+    });
+
+    it('returns a record containing the simple filters indexed by its id', () => {
+      const store = createFacetsStore([
+        createNextSimpleFilter('color', 'Red', false),
+        createNextSimpleFilter('color', 'Blue', true)
+      ]);
+
+      expect(store.getters.filtersByFacet).toEqual({
+        color: [store.state.filters['color:Red'], store.state.filters['color:Blue']]
+      });
+    });
+
+    it('returns a record containing the hierarchical filters indexed by its id', () => {
+      const store = createFacetsStore([
+        createNextHierarchicalFilter('category', 'Summer', false),
+        createNextHierarchicalFilter('category', 'Shorts', true)
+      ]);
+
+      expect(store.getters.filtersByFacet).toEqual({
+        category: [store.state.filters['category:Summer'], store.state.filters['category:Shorts']]
+      });
+    });
+
+    it('returns a record containing the number range filters indexed by its id', () => {
+      const store = createFacetsStore([
+        createNextNumberRangeFilter('price', { min: 0, max: 25 }, false),
+        createNextNumberRangeFilter('price', { min: 25, max: 50 }, true)
+      ]);
+
+      expect(store.getters.filtersByFacet).toEqual({
+        price: [store.state.filters['price:0-25'], store.state.filters['price:25-50']]
+      });
+    });
+
+    it('returns a record containing the editable number range filters indexed by its id', () => {
+      const store = createFacetsStore([
+        createNextEditableNumberRangeFilter('age', { min: null, max: 5 })
+      ]);
+
+      expect(store.getters.filtersByFacet).toEqual({
+        age: [store.state.filters['age:*-5']]
+      });
+    });
```

Comment:
I'll add it, just in case. I discussed this with @tajespasarela, and he didn't 100% convince me, I still see lots of duplicity between these tests, but I agree that one is testing mixing different filters, and the error will be harder to spot, while others could help you debug easily as you have less data in the store.